test_runner/performance/test_physical_replication.py

```diff
@@ -76,14 +79,18 @@ def test_ro_replica_lag(
             endpoint_type="read_only",
             settings={"pg_settings": {"hot_standby_feedback": "on"}},
         )
+        log.info("Replica endpoint ID: {}", cast("str", replica["endpoint"]["id"]))
         replica_env = master_env.copy()
         replica_env["PGHOST"] = replica["endpoint"]["host"]
         neon_api.wait_for_operation_to_finish(project_id)
 
-        replica_connstr = neon_api.get_connection_uri(
-            project_id,
-            endpoint_id=replica["endpoint"]["id"],
-        )["uri"]
+        replica_connstr = cast(
+            "str",
```

Comment:
IMO: these explicit type casts just do not make any sense to me in the test. When we had 4 lines, we now have 7, so almost x2 increase in this particular case. It looks ugly, it clutters code visually, so it becomes hard to read, it helps with nothing really.

Reply:
I think we should use a generator for these API bindings to help out with things.